File: 0205-isomorphic-strings/0205-isomorphic-strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        Map<Character, Character> map = new HashMap();
        
        for(int i=0; i<s.length(); i++) {
            char sC = s.charAt(i);
            char tC = t.charAt(i);
            
            if(!map.containsKey(sC)) {
                if(map.containsValue(tC))
                    return false;
                else
                    map.put(sC, tC);
            }
            else {
                if(map.get(sC) != tC)
                    return false;
            }
        }
        return true;
    }
}

File: 0205-isomorphic-strings/0205-isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        mapping = {}
        for c1, c2 in zip(s, t):
            if c1 not in mapping:
                if c2 in mapping.values():
                    return False
                mapping[c1] = c2
            else:
                if mapping[c1] != c2:
                    return False
            
        return True
